sphinx/api/types.rst

```diff
@@ -0,0 +1,315 @@
+creating it using one of the available constructors.
+
+Additionally, :py:class:`openoa.plant.PlantData` requires a metadata specification, as provided
+through the :py:class:`openoa.plant.PlantMetaData` class, which enable a series of data validations
+that run at initialization. Optionally, this can be re-run later using
+:py:class:`openoa.plant.PlantData.validate()`. Specifically, using the new
+:py:class:`openoa.plant.PlantMetaData` structure, a user can map the column names already present
+in their data to those that OpenOA will use internally, set the expected frequency of the their
+time-dependent data, and check the expected units and datatypes that the data should use. These
+configurations can be set in either a dictionary, or a metadata file using a JSON or YAML data
+format, whichever is preferable to the user. In the examples, the file "examples/data/plant_meta.yml"
+or "examples/data/plant_meta.json" are used interchangeably, and can be used as a guide.
+
+The following sections will show how each of the data should be configured, and where to check for
+these settings in the code itself. It should be noted that neither the :py:attr:`XMetaData.dtypes`, nor the
+:py:attr:`XMetaData.units` can be set manually, or updated as they are exclusively for reference to users.
+
+Each of the :py:class:`XMetaData` classes accept the inputs of the elements under the column "Field Name" in
+the following subsections, in addition to the frequency (:py:attr:`freq`) for time-dependent inputs. All
+other attributes of the metadata classes for user reference, and therefore immutable. After setting
+each of the inputs, users can access the dictionary elements :py:attr:`col_map`, :py:attr:`dtypes`, and :py:attr:`units` to
+work with the various mappings. Below, is a demonstration of this mapping in practice, showing the
+SCADA data mapping used in "examples/data/plant_meta.yml", where the keys are the OpenOA column names,
+and the values are the La Haute Borne data naming conventions. This mapping can be repeated for each
+of the other metadata types.
+
+.. literalinclude:: ../../examples/data/plant_meta.yml
+    :language: yaml
+    :lines: 35-43
+    :linenos:
+    :lineno-start: 35
+
+
+
+Data Schema User Guide
+**********************
+
+The following subsections will demonstrate the required data mapping schemas to enable
+:py:class:`openoa.plant.PlantData` to validate and and convert user-specified data to a validated
+OpenOA schema for use throughout the codebase. The data columns and their associated units and
+datatypes will be shown in a table, followed by a demonstration of how this is used in the La Haute
+Borne example data used for all of the example analysis workflows. It should be noted that the
+column "Field Name" is the internal naming convention, and should be the dictionary or JSON/YAML
+key with the actual column naming as its associated value (as is seen in the YAML snippets for each
+section).
```

Comment:
@RHammond2 do you think we should mention what to do if a user doesn't have data for a specific data column? E.g., just delete that key/value pair in the dictionary, or leave the value as None?

Reply:
I'm writing a small paragraph on this because you technically need to provide the column, but depending on how you're validating it, you don't actually need it.